odc/stats/io.py

```diff
@@ -242,13 +246,17 @@ def _ds_to_thumbnail_cog(self, ds: xr.Dataset, task: Task) -> List[Delayed]:
                                 crs=CRS.from_epsg(task.geobox.crs.to_epsg()))
 
         if task.product.preview_image_ows_style:
+            _log.info("Generate thumbnail")
             try:
-                thumbnail_cog = self._get_thumbnail(ds, task, task.product.preview_image_ows_style, input_geobox, odc_file_path)
-                thumbnail_cogs.append(thumbnail_cog)
-            except AttributeError:
-                _log.error(f"Cannot parse OWS styling: {task.product.preview_image_ows_style}.")
+                image = self._apply_color_ramp(ds, task.product.preview_image_ows_style, task.time_range.start)
+            except AttributeError as e:
+                _log.error(f"{e} Cannot parse OWS styling: {task.product.preview_image_ows_style}.")
             except ImportError as e:
                 raise type(e)(str(e) + '. Please run python -m pip install "odc-stats[ows]" to setup environment to generate thumbnail.')
+            else:
+                display_pixels = _xarray_to_list(image, task.geobox.shape[0:2])
+                thumbnail_cog = self._get_thumbnail(display_pixels, input_geobox, odc_file_path)
+                thumbnail_cogs.append(thumbnail_cog)
```

Comment:
I got question for this actually, what was the purpose? from my understanding of the original code, if ows style failed, it wouldn't generate thumbnail but continue normally for the rest, then what's the point of raising `ImportError` then? Shouldn't we raise all or ignore all?

Reply:
`ImportError` aims to display this message `run python -m pip install odc-stats[ows]`. Just in case some users may setup their odc-stats apps by `run python -m pip install odc-stats` but still try to use OWS features.